crates/shared/src/bad_token/token_owner_finder.rs

```diff
@@ -342,29 +338,22 @@ impl TokenOwnerFinding for TokenOwnerFinder {
         futures::pin_mut!(stream);
 
         while let Some(chunk) = stream.next().await {
-            let mut batch = Web3CallBatch::new(self.web3.transport().clone());
-            let futures = chunk
-                .iter()
-                .map(|&address| {
-                    let balance = instance.balance_of(address).batch_call(&mut batch);
-                    async move {
-                        let balance = match balance.await {
-                            Ok(balance) => Some(balance),
-                            Err(err) if EthcontractErrorType::is_contract_err(&err) => None,
-                            Err(err) => return Err(err),
-                        };
-
-                        Ok((address, balance))
+            let futures = chunk.into_iter().map(|owner| {
+                let call = instance.balance_of(owner).call();
+                async move {
```

Comment:
Is there a reason this follow up handling is nested in here (rather than mapping the result `from try_join_all`)

Reply:
I wanted to keep it as close to the original as possible. For `try_join_all()` to make sense we actually have to return some error case which should break the whole request batch.
Alternatively we could simply do `join_all()` and filter out those RPC errors (that would currently abort everything). Since we only need to find some owner that has at least the requested balance from the callers perspective it wouldn't really make a difference.